core/src/worker/workflow/wft_poller.rs

```diff
@@ -93,15 +100,19 @@ impl WFTPollerShared {
     /// Makes either the sticky or non-sticky poller wait pre-permit-acquisition so that we can
     /// balance which kind of queue we poll appropriately.
     pub(crate) async fn wait_if_needed(&self, is_sticky: bool) {
+        // Sticky shouldn't start polling until after the first non-sticky poll has been allowed
+        if is_sticky {
+            self.wait_for_first_nonsticky_poll.notified().await;
+        }
         // If there's a sticky backlog, prioritize it.
         if !is_sticky {
             let backlog = *self.last_seen_sticky_backlog.0.borrow();
-            if backlog > 1 {
+            if backlog >= 1 {
                 let _ = self
                     .last_seen_sticky_backlog
                     .0
                     .clone()
-                    .wait_for(|v| *v <= 1)
+                    .wait_for(|v| *v < 1)
```

Comment:
```suggestion
                    .wait_for(|v| *v == 0)
```

(I have no idea currently what's going on in this code, but I see `v` is `usize`.)